create-derivative-topic/src/create_derivative_topic/main.py

```diff
@@ -0,0 +1,41 @@
+import roboto
+
+from .logger import logger
+from .utils import get_tracking_error
+
+
+def main(context: roboto.InvocationContext) -> None:
+    logger.setLevel(context.log_level)
+
+    action_input = context.get_input()
+
+    if action_input.files:
+        logger.info("Processing %d input file(s):", len(action_input.files))
+        for file, _ in action_input.files:
+            logger.info(
+                "%sFile: %s (ID: %s)",
+                "  ",
+                file.relative_path,
+                file.file_id,
+            )
+
+            # Get the ingested position and setpoint topics
+            pos_topic = file.get_topic("vehicle_local_position")
```

Comment:
Nit: here and below, when reading example code, it helps me to see fully spelled out variable names so that its all a bit more literate and doesn't require me to do a mental translation from shorthand that I may not be familiar with (`pos` or `sp` to `position` or `setpoint`).